pyspades/master.py

```diff
@@ -128,15 +128,20 @@ def add_descriptor(self, host: str, port: int):
         self.descriptors += [descriptor]
 
     def add_client(self, desc: MasterHostDescriptor):
-        connection = self.protocol.connect(
-            MasterConnection, desc.host, desc.port, MASTER_VERSION)
+        try:
+            connection = self.protocol.connect(
+                MasterConnection, desc.host, desc.port, MASTER_VERSION)
+        except OSError:
```

Comment:
I'm not a big fan of just silently swallowing something as broad as OSError... I like the idea of unifying the reconnect and initial connect logic, but losing the ability to debug things at all is not great.

Reply:
Yeah I would do it at higher log level though because it's something end users are probably going to want to see. Ideally combined with the other log message like before?